sdk/verifiers/python/src/pipelock_aarp_verify/provenance_proof.py

```
@@ -462,10 +464,11 @@ def _verify_entry(
         if _require_str(proof["version"], "proof version") != PROOF_VERSION:
             raise ProvenanceError("proof_structure", "unsupported proof version")
-        if (
-            _require_str(proof["transform_profile_digest"], "proof profile")
-            != PROFILE_DIGEST
-        ):
+        try:
+            profile_version(
+                _require_str(proof["transform_profile_digest"], "proof profile")
+            )
+        except ProvenanceError:
             raise ProvenanceError("proof_structure", "unknown proof profile")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)